### ISSUE TITLE
Takes some time to show the correct title after switching tabs in Firefox

### ISSUE THREAD

**Issue by anupam207** [franglais125/no-title-bar#17]
I usually observe this issue after using Firefox for quite some time. I have **Replace App Menu name with Window name** option on. So the tab title is shown on the top panel. But when I switch to another tab the title of the previous tab is shown for some time before correctly showing title of the current tab.

Distro: Ubuntu GNOME 17.04, shell version 3.24.2
I noticed the same issue with Pixel Saver too.

**Comment by franglais125** [comment]
Thanks a lot for reporting this! I'll try to take a look as soon as possible.

**Comment by anupam207** [comment]
@franglais125 I don't use multiple monitors, only one laptop screen. And as I mentioned, it starts occurring after using Firefox for quite some time.

**Comment by anupam207** [comment]
Here's a screenshot. It shows the title of the previously selected Wikipedia tab, instead of the current GitHub tab.
![tab-switch](https://user-images.githubusercontent.com/20688749/29746083-da28fe44-8aeb-11e7-8209-65819387d3dd.png)
I noticed whenever the issue is present, if I **right click** on the current tab or inside the current page, it immediately starts showing the correct title.

**Comment by franglais125** [comment]
Sorry for the late reply, as I mentioned somewhere else, I'm traveling at the moment. I'll try to look into it soon.

In any case, this sounds more like a problem within Firefox rather than the extension. I certainly don't mean to sound dismissive - I'll test this myself.

Let me know if you can get more information, and I'll let you know as soon as I understand better the cause of the issue.

Cheers!

**Comment by franglais125** [comment]
I think this might have been fixed with the latest commit to master [8a1406a]. Would you mind testing the latest master branch?

**Comment by anupam207** [comment]
@franglais125 Thanks a lot for your reply. Here's what I tried:

1. I downloaded the master zip from GitHub.
2. Disabled No Title Bar from GNOME Tweaks.
3. Extracted the contents of the zip to `~/.local/share/gnome-shell/extensions/no-title-bar@franglais125.gmail.com/` replacing the old files.
4. Enabled the extension again.

But the extension threw error immediately:
![n-t-b-error](https://user-images.githubusercontent.com/20688749/32145838-c79935e2-bcf4-11e7-9c7a-4d3b57ba22f0.png)

**Comment by franglais125** [comment]
Hmm, would you mind doing the following:

- Log out and log in (so that the extension is reloaded properly).
- Check for errors in: `sudo journalctl /usr/bin/gnome-shell`

**Comment by anupam207** [comment]
@franglais125 I forgot to mention I restarted GNOME shell after activating the extension (I'm in Xorg). I'll try again and look for the error.

**Comment by franglais125** [comment]
Ugh, silly me... Just remembered, installing from the zip from github is not ok. The reason is some things need to be "compiled", like the schemas, translations, etc.

In Ubuntu:
```
git clone https://github.com/franglais125/no-title-bar
cd no-title-bar/
make install
```

And then reload. If you don't have all the utils:
`sudo apt install make git gettext`

**Comment by anupam207** [comment]
Thanks again @franglais125 . I just `git clone`d and built using `make` just as you suggested, and then logged out and logged in again. But unfortunately it's still throwing error. 

I ran `sudo journalctl /usr/bin/gnome-shell` and found this line repeated a few times:
`Extension "no-title-bar@franglais125.gmail.com" had error: Error: second argument to Lang.bind() must be a`

P.S. One more thing I forgot to mention, I have upgraded to Ubuntu 17.10, GNOME shell v3.26.1.
